crates/provider/src/fillers/nonce.rs

```diff
@@ -54,7 +54,7 @@ impl NonceManager for SimpleNonceManager {
 ///
 /// There is also an alternative implementation [`SimpleNonceManager`] that does not store the
 /// transaction count locally.
-#[derive(Clone, Debug, Default)]
+#[derive(Debug, Default)]
 pub struct CachedNonceManager {
     nonces: DashMap<Address, Arc<Mutex<u64>>>,
```

Comment:
I see, unsure how we missed this

can we instead replace this with an Arc mutex instead? not a fan of dashmap anyway

Reply:
this change we could do this immediately, not sure I fully get the other parts of the pr yet, so I suggest to submit that separately